database/src/main/scala/com/rasterfoundry/granary/ModelDao.scala

```diff
@@ -0,0 +1,43 @@
+package com.rasterfoundry.granary.database
+
+import doobie._
+import doobie.implicits._
+import doobie.postgres.implicits._
+import com.rasterfoundry.granary.datamodel.Model
+
+import java.util.UUID
+
+object ModelDao {
+  val selectF = fr"select id, name, validator, job_definition, job_queue FROM models"
+
+  def listModels: ConnectionIO[List[Model]] =
+    selectF.query[Model].to[List]
+
+  def getModel(id: UUID): ConnectionIO[Option[Model]] =
+    (selectF ++ Fragments.whereOr(fr"id = ${id}")).query[Model].option
+
+  def unsafeGetModel(id: UUID): ConnectionIO[Model] =
+    (selectF ++ Fragments.whereOr(fr"id = ${id}")).query[Model].unique
+
+  def insertModel(model: Model): ConnectionIO[Model] = {
```

Comment:
Ah yeah I agree for the most part. I was trying to avoid the whole `Foo`, `Foo.Create`, `Foo.WithX` adventure that we have over in RF. We could be clever with the decoder to allow not passing an ID, but that still sends the wrong message I think. I'll add a `Create`.

Reply:
c277bbd